### PR TITLE
allow plain paths in page_from_file

### PR DESCRIPTION
**Pull request by bertsky**
I repeatedly find myself in need of this.

BTW, there still is a problem with our tests. I keep getting [`I/O operation on closed file`](https://github.com/pytest-dev/pytest/issues/14) in test_logging.py when running in local environments. More explanations [here](https://github.com/pytest-dev/pytest/issues/5743). Also explains our previous problems with [click.CliRunner](https://github.com/pytest-dev/pytest/issues/3344) I believe. I don't know how to actually fix this, but running with `-s` helps as a workaround.

Let's see how the CI reacts.

### CODE REVIEW

**Comment by bertsky** [comment]
CI test is only macOS (still broken).

BTW, is there a reason for not exporting `OcrdPage` in `ocrd_models` directly?

**Comment by kba** [comment]
> CI test is only macOS (still broken).

Good  :+1:

> BTW, is there a reason for not exporting `OcrdPage` in `ocrd_models` directly?

No, just an oversight AFAICT.

File: ocrd_modelfactory/ocrd_modelfactory/__init__.py [hunk 1]
```diff
@@ -79,15 +82,32 @@ def page_from_image(input_file, with_tree=False):
...
             and reverse mapping, too (cf. :py:func:`ocrd_models.ocrd_page.parseEtree`)
     """
+    if not isinstance(input_file, OcrdFile):
+        mimetype = guess(input_file)
+        if mimetype is None:
+            mimetype = guess_type(input_file)[0]
+        else:
+            mimetype = mimetype.mime
+        if mimetype is None:
+            mimetype = EXT_TO_MIME.get(Path(input_file).suffix, None)
```
Comment (kba) [inline ocrd_modelfactory/ocrd_modelfactory/__init__.py:103]: Perhaps we should move this to a utility function, also to document why we need the three mechanisms (`mimetype.guess`, `filetype.guess_type` and `EXT_TO_MIME`).

**Review by kba — APPROVED** [review]
Makes sense to allow passing string paths.

File: ocrd_modelfactory/ocrd_modelfactory/__init__.py (hunk 1 above)
Comment (bertsky) [reply ocrd_modelfactory/ocrd_modelfactory/__init__.py:103]: Sure. But note that there's already a difference to what we need (regarding these detectors) in resmgr.download.

Comment (kba) [reply ocrd_modelfactory/ocrd_modelfactory/__init__.py:103]: Yes, but these differences (falling back to `application/octet-stream` vs. raising an error) can be handled with function args.

Comment (kba) [reply ocrd_modelfactory/ocrd_modelfactory/__init__.py]: https://github.com/OCR-D/core/pull/1045/commits/ed19b654d70eb9437ea1d46ce065c7471f08787e

**Pull request merged**